libraries/AP_Logger/AP_Logger.cpp

```diff
@@ -601,8 +603,9 @@ bool AP_Logger::logging_enabled() const
 }
 bool AP_Logger::logging_failed() const
 {
-    if (_next_backend < 1) {
-        // we should not have been called!
+    if (_next_backend == 0) {
```

Comment:
This should probably also check _params.backend_types != 0 - i.e. we are not failed if there weren't supposed to be any backends anyway,